tools/cache_creator/cache_creator.cpp

```diff
@@ -161,6 +162,29 @@ bool hexStringToUuid(llvm::StringRef hexStr, llvm::MutableArrayRef<uint8_t> outU
   return true;
 }
 
+llvm::Expected<ElfLlpcCacheInfo> getCacheInfoFromMetadataBlob(llvm::StringRef noteBlob) {
+  llvm::msgpack::Document *document = new llvm::msgpack::Document;
+  if (!document->readFromBlob(noteBlob, false)) {
+    return llvm::createStringError(llvm::object::object_error::invalid_file_type, "Bad PAL metadata format");
+  }
+
+  llvm::msgpack::MapDocNode pipelineNode =
+      document->getRoot().getMap(true)["amdpal.pipelines"].getArray(true)[0].getMap(true);
```

Comment:
I think this should check that each accessed node is of expected type in case the document has a different format.